rust/kernel/task.rs

```diff
@@ -124,11 +124,20 @@ impl Task {
         }
     }
 
+    /// Returns a raw pointer to the underlying C task struct.
+    ///
+    /// # Safety
+    ///
+    /// Callers must ensure that the returned pointer doesn't outlive the current task/thread.
+    pub unsafe fn as_raw(&self) -> *mut bindings::task_struct {
```

Comment:
This function should not be `unsafe`. Similar to why [`Box::into_raw`](https://doc.rust-lang.org/std/boxed/struct.Box.html#method.into_raw) is also not `unsafe`. This is because in order to use the pointer, the user will still have to write an `unsafe` block.